luci/luci-app-falter-owm/files/owm.sh

```diff
@@ -0,0 +1,321 @@
+#json_get_var timeSinceStartup timeSinceStartup
+olsr4links=""
+if json_is_a links array;then
+	json_for_each_item olsr4_links links
+fi
+json_cleanup
+json_load "$( echo /links | nc ::1 9090 2>/dev/null)" 2>/dev/null
+#json_get_var timeSinceStartup timeSinceStartup
+olsr6links=""
+if json_is_a links array;then
+	json_for_each_item olsr6_links links
+fi
+json_cleanup
+
+
+# collect board info
+json_load "$(ubus call system board)"
+json_get_var model model
+json_get_var hostname hostname
+json_get_var system system
+json_select release
+json_get_var revision revision
+json_get_var distribution distribution
+json_get_var version version
+json_select ..
+json_load "$(ubus call system info)"
+json_get_var uptime uptime
+json_get_values loads load
+
+# if file freifunk_release is available, override version and revision
+if [ -f /etc/freifunk_release ]; then
+	. /etc/freifunk_release
+	distribution="$FREIFUNK_DISTRIB_ID"
+	version="$FREIFUNK_RELEASE"
+	revision="$FREIFUNK_REVISION"
+fi
+
+# Get Sysload
+sysload=$(cat /proc/loadavg)
+load1=$(echo "$sysload" | cut -d' ' -f1)
+load5=$(echo "$sysload" | cut -d' ' -f2)
+load15=$(echo "$sysload" | cut -d' ' -f3)
+
+# Date when the firmware was build.
+kernelString=$(cat /proc/version)
+buildDate=$(echo $kernelString | cut -d'#' -f2 | cut -c 3-)
+kernelVersion=$(echo $kernelString | cut -d' ' -f3)
+
+# contact information
+uci_load freifunk
+name="$(uci_get freifunk contact name)"
+nick="$(uci_get freifunk contact nickname)"
+mail="$(uci_get freifunk contact mail)"
+phone="$(uci_get freifunk contact phone)"
+homepage="$(uci_get freifunk contact homepage)" # whitespace-separated, with single quotes, if string contains whitspace
+note="$(uci_get freifunk contact note)"
+
+# community info
+ssid="$(uci_get freifunk community ssid)"
+mesh_network="$(uci_get freifunk community mesh_network)"
+uci_owm_apis="$(uci_get freifunk community owm_api)"
+com_name="$(uci_get freifunk community name)"
+com_homepage="$(uci_get freifunk community homepage)"
+com_longitude="$(uci_get freifunk community longitude)"
+com_latitude="$(uci_get freifunk community latitude)"
+com_ssid_scheme=$(uci_get freifunk community ssid_scheme)
+com_splash_network=$(uci_get freifunk community splash_network)
+com_splash_prefix=$(uci_get freifunk community splash_prefix)
+
+
+
+###########################
+#                         #
+#  Construct JSON-string  #
+#                         #
+###########################
+
+json_init
+json_add_object freifunk
+
+	json_add_object contact
+		if [ -n "$name" ]; then json_add_string name "$name"; fi
+		if [ -n "$mail" ]; then json_add_string mail "$mail"; fi
+		if [ -n "$nick" ]; then json_add_string nickname "$nick"; fi
+		if [ -n "$phone" ]; then json_add_string phone "$phone"; fi
+		if [ -n "$homepage" ]; then json_add_string homepage "$homepage"; fi #ToDo: list of homepages
+		if [ -n "$note" ]; then json_add_string note "$note"; fi
+		# ToDo: add location-string
```

Comment:
You did not add the location string?

Reply:
In fact the location string was never sent to owm. So I removed it too. Was just like a deprecated reminder to myself.